### DISCUSSION TITLE
Rename "Import build"? (metals-vscode)

### DISCUSSION THREAD

**Discussion by phdoerfler** [scalameta/metals#2344] [Q&A]
I'd like to rename "Import build" to "Import or reload build" or something along those lines so people know that it actually does both and is thus also applicable to BSP and not just to bloop.

Thoughts?

**Comment by tgodzik** [comment]
Honestly I don't see much difference between `import` and `reload`. `reload` is a term specific to sbt, while we use `import` for gradle, sbt, maven and mill. This is why I would rather stay with the current title.

**Reply by ckipp01** [reply]
I also think it's important to keep the `Import build` wording, but it's a valid point about `Reload`. In reality, reload is actually more BSP specific, since for BSP servers it is triggering a `workspace/reload`. This is the case both for mill BSP and also sbt BSP. I'd say that Build Import is actually Bloop specific.  We may want to re-think the wording or at least the command description in the future.

**Reply by tanishiking** [reply]
I personally don't see much difference between `import` and `reload`, but I think renaming wording `Import build` might be a good idea because
- I'm concern about whether `Import build` is meaningful to metals' end-users.
  - On the one hand, for metals developers (who know the internal architecture), the wording `import build` makes sense because we know that triggers bloop to import build from the build server.
  - On the other hand, for end-users (who aren't familiar with what's happening behind the curtain during metals' startup), the wording `Import build` may not be meaningful.
    - I guess this misconception made @phdoerfler be confused.

I couldn't come up with a good alternative wording at this time, but if there're more users who want to rename "Import build", it's worth considering rename.

**Reply by tgodzik** [reply]
Maybe then `Sync` would make sense here? I think that's how it's done in Intellij, so would be familiar to a lot of users.

**Reply by kpodsiad** [reply]
I'm gonna dig it out a little :D

> Maybe then `Sync` would make sense here? I think that's how it's done in Intellij, so would be familiar to a lot of users.

Actually, in IntelliJ, the button was always called `reload` so renaming `import` to `import or reload` (or other proposition with `reload`) can be beneficial for users. For people, who aren't familiar with Metals internals `reload` may sound more intuitive - if I'm using metals for while and I made a change in the build tool why do I have to import (from scratch) instead of reloading/refreshing (only changes).

**Reply by tgodzik** [reply]
Maybe we should special case the first import and everything rename to `reload` ?
